### DISCUSSION TITLE
Android Back Button Issue

### DISCUSSION THREAD

**Discussion by piyushpandey-kiwi** [ionic-team/capacitor#3172] [Q&A]
Hi,

I am using capacitor plugin with angular 8 and when i click on back button in android then nothing works. Please suggest asap.

I am using this code in app.component.ts

 App.addListener('backButton', (res) => {
        console.log('back res', res);
});

**Comment by jcesarmobile** [comment]
what does "nothing works" mean?

**Reply by piyushpandey-kiwi** [reply]
Means its not navigating to previous url

**Reply by piyushpandey-kiwi** [reply]
App.addListener('backButton', (res) => {
console.log('back res', res);
this.location.back();
});

**Reply by jcesarmobile** [reply]
yeah, backButton listener is meant to be used when you don't want the default behavior of the back button but to handle the navigation yourself.
The default behavior is to navigate back, but if you add the backButton listener then it doesn't navigate back.

**Reply by piyushpandey-kiwi** [reply]
so how to navigate back using backButton listener. As i have to manage backbutton functionality using this code:

App.addListener('backButton', (res) => {
        console.log('back res', res);
        if (this.activeModal.hasOpenModals()) {
          this.activeModal.dismissAll();
        } else if (this.router.url === '/home/dashboard' || this.router.url === '/home/dashboard/control') {
          this.showConfirm();
        } else if (this.router.url === '/landing/forgot-password') {
          this.router.navigateByUrl('/landing/login');
        } else if (this.router.url === '/landing/login') {
          this.showConfirm();
        } else {
          this.location.back();
        }
      });
